python-sdk/src/langwatch/evaluations.py

```
@@ -142,38 +153,47 @@ def prepare_data(
 ):
     span_ctx = get_current_span().get_span_context()
-
-    data = {
+    dataDict = {
         "trace_id": format(span_ctx.trace_id, "x"),
         "span_id": format(span_ctx.span_id, "x"),
+        **(data.model_dump(exclude_unset=True, exclude_none=True) if isinstance(data, BasicEvaluateData) else data or {}),
     }
     if input is not None:
-        data["input"] = input
+        warn("For the `evaluate` or `async_evaluate` function, the `input` argument is deprecated and will be removed in a future version. Future versions of the SDK will not support it. Please use the `data` argument instead, to have complete control over the data structure.")
```

Ruff (0.8.2)

No explicit `stacklevel` keyword argument found

(B028)